src/mlpack/tests/imputation_test.cpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * @file imputation_test.cpp
+ * @author Keon Kim
+ *
+ * Tests for data::Imputer class
+ */
+#include <sstream>
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/dataset_info.hpp>
+#include <mlpack/core/data/map_policies/increment_policy.hpp>
+#include <mlpack/core/data/map_policies/missing_policy.hpp>
+#include <mlpack/core/data/imputer.hpp>
+#include <mlpack/core/data/imputation_methods/custom_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/listwise_deletion.hpp>
+#include <mlpack/core/data/imputation_methods/mean_imputation.hpp>
+#include <mlpack/core/data/imputation_methods/median_imputation.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "old_boost_test_definitions.hpp"
+
+using namespace mlpack;
+using namespace mlpack::data;
+using namespace std;
+
+BOOST_AUTO_TEST_SUITE(ImputationTest);
+
+/**
+ * Make sure a CSV is loaded correctly.
+ */
+BOOST_AUTO_TEST_CASE(DatasetMapperImputerTest)
+{
+  fstream f;
+  f.open("test_file.csv", fstream::out);
+  f << "a, 2, 3"  << endl;
+  f << "5, 6, 7"  << endl;
+  f << "8, 9, 10" << endl;
+  f.close();
+
+  arma::mat input;
+  arma::mat output;
+  string missingValue = "a";
+  double customValue = 99;
+  size_t feature = 0;
+
+  DatasetInfo info;
+  BOOST_REQUIRE(data::Load("test_file.csv", input, info) == true);
+
+  BOOST_REQUIRE_EQUAL(input.n_rows, 3);
+  BOOST_REQUIRE_EQUAL(input.n_cols, 3);
+
+  /* TODO: Connect Load with the new DatasetMapper instead of DatasetInfo*/
+
+  Imputer<double,
+          DatasetInfo,
+          CustomImputation<double>> impu(info);
+  impu.Impute(input, output, missingValue, customValue, feature);
+  // Remove the file.
```

Comment:
yea, this is the part I wanted to test the whole process of Loading, mapping, and imputation.
I will at least make a mock data and test only mapping and imputation for this.